### PR TITLE
Xeno Alliance Announcements + Greeno Civil War

### PR DESCRIPTION
**Pull request by ihatethisengine**
# About the pull request

Xenos now get messages when their queen set/break alliance to another faction and when other queen set/break alliance with their hive.

Corrupted Xenos with implanted IFF tag now has a choice to defect from the hive and become Renegade (hive allied to USCM) when Queen decides to break alliance with USCM. Xenos that stay loyal to Queen rip the IFF tags out. You have only 10 seconds to make a decision, so think quick. By default xenos stay loyal to Queen.

Renegade Hive is allied to all human factions, but it mostly affects structures and weeds. Renegade ability to attack someone fully depends on its IFF tag settings.

Please check my messages because I'm not very good at writing.

# Explain why it's good for the game

More announcements are good because it's less confusing. You know when someone set ally to you and you know when someone is betraying you. It makes sense because allied xenos share announcements anyway. And sudden betrayals are always a bit cheesy.

I think the Renegade addition makes research a little more fun and rewarding. Now, if you implant corrupted xeno with an IFF tag, the xeno player can choose to remain loyal to you if/when the Queen decides to betray. Plus corrupted xenos with the IFF tag are no longer forced to betray, so it's also good. Not sure if that makes sense lore-wise, but since corrupted are artificially created and their DNA is decrypted, it makes at least some sense. Plus we kinda have tamed xenos which work really similarly. 

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl: ihatethisengine
add: Added announcements for xenos about forming and breaking alliances.
add: Xenos with IFF tag now have a choice to stay loyal to USCM when Queen decides to betray.
/:cl:

### CODE REVIEW

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 1]
```diff
@@ -1250,6 +1252,96 @@
...
+/datum/hive_status/proc/give_deflection_choice(mob/living/carbon/xenomorph/xeno, faction)
+	if(alert(xeno, "Your Queen has broken alliance with [faction]. Your IFF tag begins to suppress your connection with the hive. Do you wish to remove the tag and stay with Queen or do you wish to stay loyal to your new masters? You have 10 seconds to decide.", "Choose", "Stay with Queen", "Stay with your masters") == "Stay with your masters")
+		if(!xeno.iff_tag)
+			to_chat(xeno, SPAN_XENOWARNING("It's too late now. Your IFF tag is broken and your service to the Queen continues."))
+			return
+		renegades += xeno
+		xeno.set_hive_and_update(XENO_HIVE_RENEGADE)
+		to_chat(xeno, SPAN_XENOANNOUNCE("You lost connection with your hive. Now there is no Queen, only masters."))
+		return
+	to_chat(xeno, SPAN_XENOWARNING("You decided to stay with Queen."))
```
[the remark this answers is not in the document]
Comment (ihatethisengine) [reply code/modules/mob/living/carbon/xenomorph/xeno_defines.dm]: Yes I intended but then changed my mind.

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 2]
```diff
@@ -1250,6 +1252,96 @@
...
+			continue
+		if(xeno in renegades)
+			continue
+		if(xeno.caste_type == XENO_CASTE_QUEEN)
+			continue
+		INVOKE_ASYNC(src, PROC_REF(give_deflection_choice), xeno, faction)
+	addtimer(CALLBACK(src, PROC_REF(handle_renegades), faction), 10 SECONDS)
+
+/datum/hive_status/proc/give_deflection_choice(mob/living/carbon/xenomorph/xeno, faction)
+	if(alert(xeno, "Your Queen has broken alliance with [faction]. Your IFF tag begins to suppress your connection with the hive. Do you wish to remove the tag and stay with Queen or do you wish to stay loyal to your new masters? You have 10 seconds to decide.", "Choose", "Stay with Queen", "Stay with your masters") == "Stay with your masters")
```
[the remark this answers is not in the document]
Comment (ihatethisengine) [reply code/modules/mob/living/carbon/xenomorph/xeno_defines.dm]: I'll be real, because I am testing everything on linux via wine tgui looks like a mess to me, but I think I got it working.

**Comment by ihatethisengine** [comment]
I decided to limit the renegade feature to corrupted, less checks for normal hive and it makes more sense for corrupted.

**Comment by ihatethisengine** [comment]
Thinking how to handle renegades allies. Thinking of three options:
1) Hardcode ally to all human factions, so renegades won't be able to participate in HvH at all.
2) Hardcode ally to «good» factions (probably CLF won't have IFF tags outside of events and during an event admins can change alliances how they want)
3) Actually check IFF tag to decide who renegade can harm and who cannot. Sounds like the best option, but is probably overkill...

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 3]
```diff
@@ -1250,6 +1252,101 @@
...
+	latejoin_burrowed = FALSE
+
+/datum/hive_status/corrupted/renegade/New()
+	. = ..()
+	hive_structures_limit[XENO_STRUCTURE_EGGMORPH] = 0
+	hive_structures_limit[XENO_STRUCTURE_EVOPOD] = 0
+	allies[FACTION_MARINE] = TRUE
+
+/datum/hive_status/corrupted/renegade/faction_is_ally(faction, ignore_queen_check = TRUE)
+	return . = ..()
```
Comment (Drulikar) [inline code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:1276]:
```suggestion
	return ..()
```
But also this currently entire proc at this moment can just be removed since all it does is call parent.

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 4]
```diff
@@ -1250,6 +1252,101 @@
...
+		if(xeno.stat == DEAD)
+			continue
+		if(!xeno.iff_tag)
+			continue
+		if(xeno in renegades)
+			continue
+		if(!(faction in xeno.iff_tag.faction_groups))
+			continue
+		xeno.visible_message(SPAN_XENOWARNING("\The [xeno] rips out [xeno.iff_tag]!"), SPAN_XENOWARNING("You rip out [xeno.iff_tag]! For the hive!"))
+		xeno.iff_tag = null
```
Comment (Drulikar) [inline code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:1342]: Isn't all of this currently redundant to `/datum/hive_status/corrupted/proc/give_defection_choice` ? As in that proc will null out their iff tag and give a visible message for every defector?

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm (hunk 3 above)
Comment (ihatethisengine) [reply code/modules/mob/living/carbon/xenomorph/xeno_defines.dm]: Parent has ignore_queen_check = FALSE, so alliance doesn't work because they have no queen

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm (hunk 4 above)
Comment (ihatethisengine) [reply code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:1342]: Sadly timed tgui_alert doesn't work for mobs w/o client. Also Queen removing her IFF tag also being handled here (she cannot become renegade obv).

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm (hunk 3 above)
Comment (Drulikar) [reply code/modules/mob/living/carbon/xenomorph/xeno_defines.dm]: This is not a very intuitive way to do this but I guess.

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm (hunk 4 above)
Comment (Drulikar) [reply code/modules/mob/living/carbon/xenomorph/xeno_defines.dm]: Then at least you should null check the iff tag to perform the message else they will in chat double rip out the same tag.

Comment (Drulikar) [reply code/modules/mob/living/carbon/xenomorph/xeno_defines.dm]: nevermind thats the first check now.

**Comment by ihatethisengine** [comment]
> Thinking how to handle renegades' allies. Thinking of three options:
> 
> 1. Hardcode ally to all human factions, so renegades won't be able to participate in HvH at all.
> 2. Hardcode ally to «good» factions (probably CLF won't have IFF tags outside of events and during an event admins can change alliances how they want)
> 3. Actually check IFF tag to decide who renegade can harm and who cannot. Sounds like the best option, but is probably overkill...

So what I did:
Renegade Hive is allied to all humanoids by default. Mostly it affects structures (weeds won't slowdown humans etc.)
Renegade Xeno ability to attack other xenos or humans depends on its IFF tag setting. You cannot attack a xeno that shares at least one IFF tag setting with you. Otherwise you can, even if the xeno is from the same hive. Same with humans. If you somehow lose your IFF tag, you can basically attack anyone. This way renegades are very flexible and we can get pretty cool situations.

**Comment by ihatethisengine** [comment]
Now thinking how to best handle renegades health. When you defect, chances are there are no drones with you and you might be already wounded and not able to de-evolve to drone to make weeds. Thinking of following options:
1) Allow renegades (or maybe just xenos in general) to de-evolve regardless of health (or require to have half of HP instead of max).
2) Give temporary regen off weeds, so you can disengage, prepare resting grounds and whatever.
3) Give constant regen off weeds. Makes things more dynamic for renegades and allows even one renegade to have some influence on the round. Probably makes no sense and OP, but sounds interesting.
4) Give research some way to deal with it, maybe some chem that can heal xenos (maybe there is something already).

**Comment by ihatethisengine** [comment]
Well, let's try this... Will drop this change if too much.

**Comment by ihatethisengine** [comment]
Nah let's players handle this, at least for now... Apparently it's not hard for research to make a chem that can heal xenos.

File: code/modules/mob/living/carbon/xenomorph/damage_procs.dm [hunk 5]
```diff
@@ -29,6 +29,8 @@
 		user.put_in_hands(iff_tag)
 		iff_tag = null
 		user.visible_message(SPAN_NOTICE("[user] removes \the [src]'s IFF tag."), SPAN_NOTICE("You remove \the [src]'s IFF tag."), max_distance = 3)
+		if(src.hive.hivenumber == XENO_HIVE_RENEGADE) //it's important to know their IFF settings for renegade
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/damage_procs.dm:32]:
```suggestion
		if(hive.hivenumber == XENO_HIVE_RENEGADE) //it's important to know their IFF settings for renegade
```

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 6]
```diff
@@ -1181,6 +1184,7 @@
 	destruction_allowed = XENO_NOBODY
 	dynamic_evolution = FALSE
 	allow_no_queen_actions = TRUE
+	allow_no_queen_evo = TRUE
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:1187]: why are all these being added when this isn't an override

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 7]
```diff
@@ -1296,6 +1303,124 @@
...
+	if(tgui_alert(xeno, "Your Queen has broken alliance with [faction]. Your IFF tag begins to suppress your connection with the hive. Do you wish to remove the tag and stay with Queen or do you wish to stay loyal to your new masters? You have 10 seconds to decide.", "Choose", list("Stay with Queen", "Stay with your masters"), 10 SECONDS) == "Stay with your masters")
+		if(!xeno.iff_tag)
+			to_chat(xeno, SPAN_XENOWARNING("It's too late now. Your IFF tag is broken and your service to the Queen continues."))
+			return
+		defectors += xeno
+		xeno.set_hive_and_update(XENO_HIVE_RENEGADE)
+		to_chat(xeno, SPAN_XENOANNOUNCE("You lost connection with your hive. Now there is no Queen, only masters."))
+		to_chat(xeno, SPAN_NOTICE("Your IFF tag settings are: [english_list(xeno.iff_tag.faction_groups, "None")]. You cannot attack creatures within these settings."))
+		return
+	xeno.visible_message(SPAN_XENOWARNING("\The [xeno] rips out [xeno.iff_tag]!"), SPAN_XENOWARNING("You rip out [xeno.iff_tag]! For the hive!"))
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:1401]:
```suggestion
	xeno.visible_message(SPAN_XENOWARNING("[xeno] rips out [xeno.iff_tag]!"), SPAN_XENOWARNING("You rip out [xeno.iff_tag]! For the hive!"))
```

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 8]
```diff
@@ -1296,6 +1303,124 @@
...
+
+/datum/hive_status/corrupted/proc/handle_defectors(faction)
+	for(var/mob/living/carbon/xenomorph/xeno in totalXenos)
+		if(!xeno.iff_tag)
+			continue
+		if(xeno in defectors)
+			continue
+		if(!(faction in xeno.iff_tag.faction_groups))
+			continue
+		xeno.visible_message(SPAN_XENOWARNING("\The [xeno] rips out [xeno.iff_tag]!"), SPAN_XENOWARNING("You rip out [xeno.iff_tag]! For the hive!"))
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:1414]:
```suggestion
		xeno.visible_message(SPAN_XENOWARNING("[xeno] rips out [xeno.iff_tag]!"), SPAN_XENOWARNING("You rip out [xeno.iff_tag]! For the hive!"))
```

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 9]
```diff
@@ -1296,6 +1303,124 @@
...
+		xeno.adjustBruteLoss(50)
+		xeno.iff_tag.forceMove(get_turf(xeno))
+		xeno.iff_tag = null
+	if(!length(defectors))
+		return
+
+	xeno_message(SPAN_XENOANNOUNCE("You sense that [defectors.Join(", ")] turned their backs against their sisters and the Queen in favor of their slavemasters!"), 3, hivenumber)
+	defectors = list()
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:1422]:
```suggestion
	defectors.Cut()
```

File: code/modules/mob/living/carbon/xenomorph/xeno_verbs.dm [hunk 10]
```diff
@@ -25,6 +25,13 @@
 	if(!hive)
 		return
 
+	if(hive.hivenumber == XENO_HIVE_RENEGADE) //Renegade's ability to attack someone depends on IFF settings, not on alliance
+		if(!src.iff_tag)
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/xeno_verbs.dm:29]:
```suggestion
		if(!iff_tag)
```

File: code/modules/mob/living/carbon/xenomorph/xeno_verbs.dm [hunk 11]
```diff
@@ -25,6 +25,13 @@
...
+			to_chat(src, SPAN_NOTICE("You have no IFF tag. You can attack anyone."))
+			return
+		to_chat(src, SPAN_NOTICE("Your IFF tag settings are: [english_list(src.iff_tag.faction_groups, "None")]. You cannot attack creatures within these settings."))
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/xeno_verbs.dm:32]:
```suggestion
		to_chat(src, SPAN_NOTICE("Your IFF tag settings are: [english_list(iff_tag.faction_groups, "None")]. You cannot attack creatures within these settings."))
```

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm (hunk 6 above)
Comment (ihatethisengine) [reply code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:1187]: It is?

Comment (harryob) [reply code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:1187]: > 
> It is?

lmao my bad i got mixed up with the var you changed the comment of

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 12]
```diff
@@ -1296,6 +1303,124 @@
...
+		if(target_hive.name != faction)
+			continue
+		if(!target_hive.living_xeno_queen && !target_hive.allow_no_queen_actions)
+			return
+		if(allies[faction])
+			xeno_message(SPAN_XENOANNOUNCE("You sense that [name] Queen set up alliance with us!"), 3, target_hive.hivenumber)
+			return
+		else
+			xeno_message(SPAN_XENOANNOUNCE("You sense that [name] Queen broke alliance with us!"), 3, target_hive.hivenumber)
+			return
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:1370]:
```suggestion
			return

		xeno_message(SPAN_XENOANNOUNCE("You sense that [name] Queen broke alliance with us!"), 3, target_hive.hivenumber)
```

no need to bother with an else if the other condition returns

**Review by harryob — CHANGES_REQUESTED** [review]
nit, should be good

**Review by harryob — APPROVED** [review]

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 13]
```diff
@@ -1296,6 +1303,123 @@
...
+/datum/hive_status/corrupted/renegade/faction_is_ally(faction, ignore_queen_check = TRUE)
+	return ..()
+
+/datum/hive_status/proc/on_stance_change(faction)
+	if(!living_xeno_queen)
+		return
+	if(allies[faction])
+		xeno_message(SPAN_XENOANNOUNCE("Your Queen set up alliance with [faction]!"), 3, hivenumber)
+	else
+		xeno_message(SPAN_XENOANNOUNCE("Your Queen broke alliance with [faction]!"), 3, hivenumber)
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:1357]:
```suggestion
		xeno_message(SPAN_XENOANNOUNCE("Your Queen set up an alliance with [faction]!"), 3, hivenumber)
	else
		xeno_message(SPAN_XENOANNOUNCE("Your Queen broke the alliance with [faction]!"), 3, hivenumber)
```

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 14]
```diff
@@ -1296,6 +1303,123 @@
...
+		var/datum/hive_status/target_hive = GLOB.hive_datum[number]
+		if(target_hive.name != faction)
+			continue
+		if(!target_hive.living_xeno_queen && !target_hive.allow_no_queen_actions)
+			return
+		if(allies[faction])
+			xeno_message(SPAN_XENOANNOUNCE("You sense that [name] Queen set up alliance with us!"), 3, target_hive.hivenumber)
+			return
+
+		xeno_message(SPAN_XENOANNOUNCE("You sense that [name] Queen broke alliance with us!"), 3, target_hive.hivenumber)
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:1369]:
```suggestion
			xeno_message(SPAN_XENOANNOUNCE("You sense that [name] Queen set up an alliance with us!"), 3, target_hive.hivenumber)
			return

		xeno_message(SPAN_XENOANNOUNCE("You sense that [name] Queen broke the alliance with us!"), 3, target_hive.hivenumber)
```

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 15]
```diff
@@ -1296,6 +1303,123 @@
...
+			continue
+		if(xeno in defectors)
+			continue
+		if(xeno.caste_type == XENO_CASTE_QUEEN)
+			continue
+		INVOKE_ASYNC(src, PROC_REF(give_defection_choice), xeno, faction)
+	addtimer(CALLBACK(src, PROC_REF(handle_defectors), faction), 11 SECONDS)
+
+/datum/hive_status/corrupted/proc/give_defection_choice(mob/living/carbon/xenomorph/xeno, faction)
+	if(tgui_alert(xeno, "Your Queen has broken alliance with [faction]. Your IFF tag begins to suppress your connection with the hive. Do you wish to remove the tag and stay with Queen or do you wish to stay loyal to your new masters? You have 10 seconds to decide.", "Choose", list("Stay with Queen", "Stay with your masters"), 10 SECONDS) == "Stay with your masters")
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:1391]:
```suggestion
	if(tgui_alert(xeno, "Your Queen has broken the alliance with [faction]. Your IFF tag begins to suppress your connection with the Hive. Do you wish to remove the tag and stay with your Queen or do you wish to stay loyal to your new masters? You have 10 seconds to decide.", "Choose", list("Stay with the Queen", "Stay with your masters"), 10 SECONDS) == "Stay with your masters")
```

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 16]
```diff
@@ -1296,6 +1303,123 @@
...
+		if(!xeno.iff_tag)
+			to_chat(xeno, SPAN_XENOWARNING("It's too late now. Your IFF tag is broken and your service to the Queen continues."))
+			return
+		defectors += xeno
+		xeno.set_hive_and_update(XENO_HIVE_RENEGADE)
+		to_chat(xeno, SPAN_XENOANNOUNCE("You lost connection with your hive. Now there is no Queen, only masters."))
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:1397]:
```suggestion
		to_chat(xeno, SPAN_XENOANNOUNCE("You lost the connection with your Hive. Now you have no Queen, only your masters."))
```

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 17]
```diff
@@ -1296,6 +1303,123 @@
...
+		to_chat(xeno, SPAN_NOTICE("Your IFF tag settings are: [english_list(xeno.iff_tag.faction_groups, "None")]. You cannot attack creatures within these settings."))
+		return
+	xeno.visible_message(SPAN_XENOWARNING("[xeno] rips out [xeno.iff_tag]!"), SPAN_XENOWARNING("You rip out [xeno.iff_tag]! For the hive!"))
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:1400]:
```suggestion
	xeno.visible_message(SPAN_XENOWARNING("[xeno] rips out [xeno.iff_tag]!"), SPAN_XENOWARNING("You rip out [xeno.iff_tag]! For the Hive!"))
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by Drulikar** [comment]
Just for future reference: 
image

**Comment by ihatethisengine** [comment]
> Just for future reference: image

ohhhh was wondering what does this button do lol

File: code/modules/mob/living/carbon/xenomorph/items/iff_tag.dm [hunk 18]
```diff
@@ -19,6 +19,8 @@
 		injector.visible_message(SPAN_NOTICE("[injector] forces \the [src] into [xeno]'s carapace!"), SPAN_NOTICE("You force \the [src] into [xeno]'s carapace!"))
 		xeno.iff_tag = src
 		injector.drop_inv_item_to_loc(src, xeno)
+		if(xeno.hive.hivenumber == XENO_HIVE_RENEGADE) //it's important to know their IFF settings for renegade
+			to_chat(xeno, SPAN_NOTICE("With the insertion of the device inside your carapace your instincts have changed, you seem compelled to protect [english_list(faction_groups, "no one")]."))
```
Comment (Drulikar) [inline code/modules/mob/living/carbon/xenomorph/items/iff_tag.dm:23]:
```suggestion
			to_chat(xeno, SPAN_NOTICE("With the insertion of the device into your carapace, your instincts have changed compelling you to protect [english_list(faction_groups, "no one")]."))
```

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 19]
```diff
@@ -1296,6 +1303,123 @@
...
+			continue
+		if(xeno in defectors)
+			continue
+		if(xeno.caste_type == XENO_CASTE_QUEEN)
+			continue
+		INVOKE_ASYNC(src, PROC_REF(give_defection_choice), xeno, faction)
+	addtimer(CALLBACK(src, PROC_REF(handle_defectors), faction), 11 SECONDS)
+
+/datum/hive_status/corrupted/proc/give_defection_choice(mob/living/carbon/xenomorph/xeno, faction)
+	if(tgui_alert(xeno, "Your Queen has broken the alliance with [faction]. The talls made device inside your carapace begins to suppress your connection with the Hive. Do you wish to remove the tag and stay with your Queen or do you wish to stay loyal to your new masters? You have 10 seconds to decide.", "Choose", list("Stay with the Queen", "Stay with your masters"), 10 SECONDS) == "Stay with your masters")
```
Comment (Drulikar) [inline code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:1391]:
```suggestion
	if(tgui_alert(xeno, "Your Queen has broken the alliance with [faction]. The device inside your carapace begins to suppress your connection with the Hive. Do you wish to remove the tag and stay with your Queen or do you wish to stay loyal to your new masters? You have 10 seconds to decide.", "Choose", list("Stay with the Queen", "Stay with your masters"), 10 SECONDS) == "Stay with your masters")
```
Tall would probably be more correct, but it still sounds really weird.

File: code/modules/mob/living/carbon/xenomorph/xeno_verbs.dm [hunk 20]
```diff
@@ -25,6 +25,13 @@
 	if(!hive)
 		return
 
+	if(hive.hivenumber == XENO_HIVE_RENEGADE) //Renegade's ability to attack someone depends on IFF settings, not on alliance
+		if(!iff_tag)
+			to_chat(src, SPAN_NOTICE("You are not oblidged to protect anyone."))
```
Comment (Drulikar) [inline code/modules/mob/living/carbon/xenomorph/xeno_verbs.dm:30]:
```suggestion
			to_chat(src, SPAN_NOTICE("You are not obligated to protect anyone."))
```
obliged can work too, but its not a very common usage of the word thus it being consider archaic: 
image

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 21]
```diff
@@ -1296,6 +1303,123 @@
...
+			continue
+		if(xeno in defectors)
+			continue
+		if(xeno.caste_type == XENO_CASTE_QUEEN)
+			continue
+		INVOKE_ASYNC(src, PROC_REF(give_defection_choice), xeno, faction)
+	addtimer(CALLBACK(src, PROC_REF(handle_defectors), faction), 11 SECONDS)
+
+/datum/hive_status/corrupted/proc/give_defection_choice(mob/living/carbon/xenomorph/xeno, faction)
+	if(tgui_alert(xeno, "Your Queen has broken the alliance with [faction]. The device inside your carapace begins to suppress your connection with the Hive. Do you wish to remove the tag and stay with your Queen or do you wish to stay loyal to your new masters? You have 10 seconds to decide.", "Choose", list("Stay with the Queen", "Stay with your masters"), 10 SECONDS) == "Stay with your masters")
```
Comment (Drulikar) [inline code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:1391]: Also at the moment, the message and options are too wordy for the tgui alert:
image

Maybe this instead?
```suggestion
	if(tgui_alert(xeno, "Your Queen has broken the alliance with the [faction]. The device inside your carapace begins to suppress your connection with the Hive. Do you remove it and stay loyal to her?", "Alliance broken!", list("Stay loyal", "Obey the talls"), 10 SECONDS) == "Obey the talls")
```
image

**Review by Drulikar — APPROVED** [review]
image

**Comment by Hopekz** [comment]
tbh I don't like that half the hive can just choose to abandon the queen when they're supposed to enact the queen's will no matter what
